tycho-core/src/main/java/org/eclipse/tycho/core/maven/MavenDependencyInjector.java

```diff
@@ -151,28 +159,39 @@ private Dependency createSystemScopeDependency(ArtifactDescriptor descriptor, St
             String type = dependencyDescriptor.getType();
             if (type != null && !type.isBlank()) {
                 dependency.setType(type);
+            } else {
+                dependency.setType(descriptor.getKey().getType());
             }
+            // TODO currently we still need this to be system scoped as otherwise maven tries to resolve them
+            // and we are currently not able to intercept this see https://github.com/eclipse/tycho/issues/585
+            //
+            // We might be able to mitigate this for dependencies explicitly marked as being available from maven
+            // see https://github.com/eclipse/tycho/issues/574
+            //
             dependency.setScope(Artifact.SCOPE_SYSTEM);
-            dependency.setSystemPath(location.getAbsolutePath());
+            if (Artifact.SCOPE_SYSTEM.equals(dependency.getScope())) {
+                dependency.setSystemPath(location.getAbsolutePath());
+            }
         } else {
             ArtifactKey artifactKey = descriptor.getKey();
             dependency.setGroupId(groupId);
             dependency.setArtifactId(artifactKey.getId());
             dependency.setVersion(artifactKey.getVersion());
             dependency.setScope(Artifact.SCOPE_SYSTEM);
+            dependency.setType(artifactKey.getType());
```

Comment:
Here we propagate the actual artifact-type instead of the default (jar) this seems more semantic to me and we should just register a ArtifactHandler for the Tycho types (maybe in a later change, depending on how it is required for system-scoped items).